src/idl_gen_kotlin.cpp

```diff
@@ -1418,6 +1456,14 @@ class KotlinGenerator : public BaseGenerator {
     code += "}";
   }
 
+  bool IsScalarField(const FieldDef &field) const {
```

Comment:
Maybe make `IsScalarField()` and `IsScalarOptionalField()` methods members of the `FieldDef` class?
What do you think?
It can be re-used by other code generators (at least C++).

Reply:
I don't mind doing the move. @aardappel ?

Reply:
sgtm!